fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biomes/InternalBiomeData.java

```diff
@@ -0,0 +1,183 @@
+	private static final Map<Biome, WeightedBiomePicker> OVERWORLD_SHORE_MAP = new HashMap<>();
+	private static final Map<Biome, WeightedBiomePicker> OVERWORLD_EDGE_MAP = new HashMap<>();
+	private static final Map<Biome, VariantTransformer> OVERWORLD_VARIANT_TRANSFORMERS = new HashMap<>();
+
+	private static final Map<Biome, Biome> OVERWORLD_RIVER_MAP = new HashMap<>();
+	private static final List<Biome> OVERWORLD_INJECTED_BIOMES = new ArrayList<>();
+	private static final Set<Biome> SPAWN_BIOMES = new HashSet<>();
+
+	public static void addOverworldBaseBiome(OverworldClimate climate, Biome biome, double weight) {
+		Preconditions.checkArgument(climate != null && biome != null, "One or both arguments are null");
+		Preconditions.checkArgument(!Double.isNaN(weight), "Weight is NaN");
+		Preconditions.checkArgument(weight > 0, "Weight is zero or negative (must be positive)");
+		OVERWORLD_BASE_MODDED_BIOME_PICKERS.computeIfAbsent(climate, k -> new WeightedBiomePicker()).addBiome(biome, weight);
+		OVERWORLD_INJECTED_BIOMES.add(biome);
+	}
+
+	public static void addOverworldHillsBiome(Biome parent, Biome hills, double weight) {
+		Preconditions.checkArgument(parent != null && hills != null, "One or both arguments are null");
+		Preconditions.checkArgument(!Double.isNaN(weight), "Weight is NaN");
+		Preconditions.checkArgument(weight > 0, "Weight is zero or negative (must be positive)");
+		OVERWORLD_HILLS_MAP.computeIfAbsent(parent, biome -> DefaultHillsData.injectDefaultHills(parent, new WeightedBiomePicker())).addBiome(hills, weight);
+		OVERWORLD_INJECTED_BIOMES.add(hills);
+	}
+
+	public static void addOverworldShoreBiome(Biome parent, Biome shore, double weight) {
+		Preconditions.checkArgument(parent != null && shore != null, "One or both arguments are null");
+		Preconditions.checkArgument(!Double.isNaN(weight), "Weight is NaN");
+		Preconditions.checkArgument(weight > 0, "Weight is zero or negative (must be positive)");
+		OVERWORLD_SHORE_MAP.computeIfAbsent(parent, biome -> new WeightedBiomePicker()).addBiome(shore, weight);
+		OVERWORLD_INJECTED_BIOMES.add(shore);
+	}
+
+	public static void addOverworldEdgeBiome(Biome parent, Biome edge, double weight) {
+		Preconditions.checkArgument(parent != null && edge != null, "One or both arguments are null");
+		Preconditions.checkArgument(!Double.isNaN(weight), "Weight is NaN");
+		Preconditions.checkArgument(weight > 0, "Weight is zero or negative (must be positive)");
+		OVERWORLD_EDGE_MAP.computeIfAbsent(parent, biome -> new WeightedBiomePicker()).addBiome(edge, weight);
+		OVERWORLD_INJECTED_BIOMES.add(edge);
+	}
+
+	public static void addOverworldBiomeReplacement(Biome replaced, Biome variant, double chance, OverworldClimate[] climates) {
+		Preconditions.checkArgument(replaced != null && variant != null, "One or both arguments are null");
+		Preconditions.checkArgument(chance > 0 && chance <= 1, "Chance is not greater than 0 or less than or equal to 1");
```

Comment:
```suggestion
		Preconditions.checkArgument(chance > 0.0, "Chance is less than or equal to 0.0 (%s)", chance);
		Preconditions.checkArgument(chance <= 1.0, "Chance is more than 1.0 (%s)", chance);
```

Reply:
Just do one check and say "out of bounds"

Reply:
and not say what the bounds are? that's very helpful lmao